docs/FlowClassifyHook.md

```diff
@@ -0,0 +1,253 @@
+    uint32_t compartment_id; ///< Network compartment Id.
+    uint64_t interface_luid; ///< Interface LUID.
+    uint8_t direction;       ///< 0 = inbound, 1 = outbound
+    uint64_t flow_id;        ///< WFP flow handle
+    uint32_t state;          ///< State of the flow.
+    uint8_t* data_start;     ///< Pointer to start of stream segment data
+    uint8_t* data_end;       ///< Pointer to end of stream segment data
+} bpf_flow_classify_t;
+```
+
+### Hooks
+
+1. **new_flow_classify**
+   - **Invocation:** Invoked when a new TCP flow is established (at the WFP flow established layer).
+   - **Purpose:** Allows the eBPF program to make an initial classification decision based on connection metadata (addresses, ports, process, etc.) before any stream data is seen.
+   - **Return values:**
+     - `FLOW_CLASSIFY_ALLOW`: The flow is permitted and will not be further inspected at the stream layer; no additional hook invocations for this flow.
+     - `FLOW_CLASSIFY_NEED_MORE_DATA`: The flow requires inspection at the stream layer; subsequent TCP segments will trigger `stream_flow_classify` invocations.
+   - **Notes:** This avoids unnecessary stream processing for flows that do not require data inspection at the stream layer.
+
+2. **stream_flow_classify**
+   - **Invocation:** Called for each TCP segment on flows that were marked as needing more data by `new_flow_classify`.
+   - **Purpose:** Allows the eBPF program to inspect stream data and allow/block the connection, or request more data if classification is not yet possible.
+   - **Return values:**
+     - `FLOW_CLASSIFY_ALLOW`: The flow is permitted; no further stream inspection for this flow.
+     - `FLOW_CLASSIFY_BLOCK`: The flow is blocked; all further data is dropped (using WFP ABSORB), and the connection will be terminated or time out.
+     - `FLOW_CLASSIFY_NEED_MORE_DATA`: The current segment is allowed, but the program will be invoked again for subsequent segments until a final decision is made.
+   - **Notes:** This is only invoked for each flow segment until allow/block are returned.
+
+3. **flow_deleted**
+   - **Invocation:** Called when a flow is deleted (e.g., connection teardown) and the last action for the flow was `FLOW_CLASSIFY_NEED_MORE_DATA` (i.e., the program never returned a final allow/block decision).
+   - **Purpose:** Allows the eBPF program to perform cleanup of any per-flow state (e.g., in maps) that was maintained during inspection.
+   - **Return value:**
+     - Ignored
+   - **Notes:** This hook is only invoked for flows that were still being actively classified and not for those that were allowed immediately.
+
+```c
+/*
+ * @brief Select flows for stream layer classification
+ *
+ * Program type: \ref EBPF_PROGRAM_TYPE_FLOW_CLASSIFY
+ *
+ * Attach type(s):
+ * \ref EBPF_ATTACH_TYPE_NEW_FLOW_CLASSIFY
+ *
+ * @param[in] context \ref bpf_flow_classify_t
+ * @return ALLOW to ignore(allow) this flow, NEED_MORE_DATA for stream layer classification.
```

Comment:
```suggestion
 * @retval ALLOW The flow should be allowed.
 * @retval NEED_MORE_DATA More data is needed to determine whether the flow should be allowed.
```

Use retval not return, to document specific return values.
Also, why is BLOCK not allowed here?